From 9ef90e78ec0d2ee2b496765767ea0b08632c1b3d Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <bbhushan2@marvell.com>
Date: Tue, 29 Oct 2019 12:02:40 +0530
Subject: [PATCH 390/767] dt-bindings: perf: arm-smmuv3-pmu: Add documentation
 for arm-smmuv3 pmu

commit 1893fed3b940f7d23792dd0da2bd8c8f7acaeac9 from
git@git.assembla.com:cavium/WindRiver.linux.git

Add binding documentation for the ARM-SMMUv3 performance monitor unit.

Change-Id: I80e708ba5c466a604d2ca2fdad9afd7e428fbab4
Signed-off-by: Bharat Bhushan <bbhushan2@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/17959
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../bindings/perf/arm_smmuv3_pmu.txt          | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/perf/arm_smmuv3_pmu.txt

diff --git a/Documentation/devicetree/bindings/perf/arm_smmuv3_pmu.txt b/Documentation/devicetree/bindings/perf/arm_smmuv3_pmu.txt
new file mode 100644
index 000000000000..39c95d5d03ef
--- /dev/null
+++ b/Documentation/devicetree/bindings/perf/arm_smmuv3_pmu.txt
@@ -0,0 +1,34 @@
+* ARM SMMU-V3 PMU (Performance Monitor Extension)
+
+Required properties:
+
+- compatible:	(standard compatible string) should be:
+		"arm,smmu-pmu-v3"
+
+- reg: 		(standard registers property) physical address and
+		size of the configuration registers blocks.
+
+		SMMU-V3 PMU can have one or more Performance monitor
+		counter Group (PMCG). Each PMCG have one 4K page and
+		additional optional one more 4K page
+
+- msi-parent: 	See the generic MSI binding described in
+		devicetree/bindings/interrupt-controller/msi.txt
+		for a description of the msi-parent property.
+Example:
+
+	smmupmcg0@30100000 {
+		compatible = "arm,smmu-pmu-v3";
+
+		/* Single configuration register page */
+		reg = <0x30100000 0x1000>;
+		msi-parent = <&its 0x10>;
+	};
+
+	smmupmcg0@830000100000 {
+		compatible = "arm,smmu-pmu-v3";
+
+		/* Additional optional configuration register page */
+		reg = <0x30100000 0x1000>, <0x30110000 0x1000>;
+		msi-parent = <&its 0x11>;
+	};
-- 
2.31.1

